File: easy/543. Diameter of Binary Tree/solution.java
/**
 * Definition for a binary tree node. public class TreeNode { int val; TreeNode
 * left; TreeNode right; TreeNode(int x) { val = x; } }
 */
class Solution {
    public int diameterOfBinaryTree(TreeNode root) {

        int[] res = { 0 };
        int i = traversal(root, res);
        return res[0];
    }

    private int traversal(TreeNode root, int[] res) {

        if (root == null) {
            return 0;
        }
        int left = traversal(root.left, res);
        int right = traversal(root.right, res);
        res[0] = Math.max(res[0], left + right);
        return Math.max(left, right) + 1;
    }

}

File: easy/543. Diameter of Binary Tree/solution.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def diameterOfBinaryTree(self, root: Optional[TreeNode]) -> int:
        self.diameter = 0

        def maxDepth(curr):

            if not curr:
                return 0
            
            if not curr.left and not curr.right:
                return 1
            
            left = maxDepth(curr.left)
            right = maxDepth(curr.right)

            self.diameter = max(self.diameter, left + right)

            return 1 + max(left, right)
        
        maxDepth(root)

        return self.diameter